### ISSUE TITLE
slurm barfs on unset LFS

### ISSUE THREAD

**Issue by andre-merzky** [radical-cybertools/radical.pilot#1800]
(repotrted by Ray:

the agent_0.err shows:
```
2019-02-01 10:01:46,829: radical.saga        : MainProcess                     : MainThread     : WARNING : cannot initialize context - no session: {'LifeTime' : '-1', 'Type' : 'ssh', 'UserCert' : '/home1/06078/tg853774/.ssh/id_rsa.pub', 'UserKey' : '/home1/06078/tg853774/.ssh/id_rsa'}
Traceback (most recent call last):
  File "/work/06078/tg853774/stampede2/radical.pilot.sandbox/rp.session.js-169-25.jetstream-cloud.org.yren.017928.0000/pilot.0000/rp_install/bin/radical-pilot-agent", line 71, in <module>
    bootstrap_3(sys.argv[1])
  File "/work/06078/tg853774/stampede2/radical.pilot.sandbox/rp.session.js-169-25.jetstream-cloud.org.yren.017928.0000/pilot.0000/rp_install/bin/radical-pilot-agent", line 42, in bootstrap_3
    if agent_name == 'agent_0': agent = rpa.Agent_0(agent_name)
  File "/work/06078/tg853774/stampede2/radical.pilot.sandbox/rp.session.js-169-25.jetstream-cloud.org.yren.017928.0000/pilot.0000/rp_install/lib/python2.7/site-packages/radical/pilot/agent/agent_0.py", line 118, in __init__
    session=self._session)
  File "/work/06078/tg853774/stampede2/radical.pilot.sandbox/rp.session.js-169-25.jetstream-cloud.org.yren.017928.0000/pilot.0000/rp_install/lib/python2.7/site-packages/radical/pilot/agent/rm/base.py", line 229, in create
    return impl(cfg, session)
  File "/work/06078/tg853774/stampede2/radical.pilot.sandbox/rp.session.js-169-25.jetstream-cloud.org.yren.017928.0000/pilot.0000/rp_install/lib/python2.7/site-packages/radical/pilot/agent/rm/slurm.py", line 20, in __init__
    LRMS.__init__(self, cfg, session)
  File "/work/06078/tg853774/stampede2/radical.pilot.sandbox/rp.session.js-169-25.jetstream-cloud.org.yren.017928.0000/pilot.0000/rp_install/lib/python2.7/site-packages/radical/pilot/agent/rm/base.py", line 110, in __init__
    self._configure()
  File "/work/06078/tg853774/stampede2/radical.pilot.sandbox/rp.session.js-169-25.jetstream-cloud.org.yren.017928.0000/pilot.0000/rp_install/lib/python2.7/site-packages/radical/pilot/agent/rm/slurm.py", line 80, in _configure
    lfs_path = os.path.expandvars(lfs_path)
  File "/work/06078/tg853774/stampede2/radical.pilot.sandbox/ve.xsede.stampede2_ssh.0.50.21/lib/python2.7/posixpath.py", line 287, in expandvars
    if '$' not in path:
TypeError: argument of type 'NoneType' is not iterable
```

@iparask , @vivek-bala : can either one of you take this one, please?

**Comment by iparask** [comment]
Yes, this is because we removed the `if`s that PR #1776. I will fix it and create a PR.

**Comment by andre-merzky** [comment]
Thanks Ioannis!

**Comment by shantenujha** [comment]
hi @andre-merzky --- checking on the status of this issue please?

**Comment by andre-merzky** [comment]
@iparask : any news? :-)  Let me know if you want me to take this one.  Thanks!

**Comment by iparask** [comment]
PR #1809 resolves this.

**Comment by iparask** [comment]
Apparently the pilot launcher does in [lines 808-809](https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/pmgr/launching/default.py#L808-809):

```python
        lfs_path_per_node       = rcfg.get('lfs_path_per_node', None)
        lfs_size_per_node       = rcfg.get('lfs_size_per_node',  0)
```
I propose to change the first get to `''` instead of `None` and also shield RM from a `None` value.

**Comment by andre-merzky** [comment]
Thanks  @iparask - this is now released as v0.50.22.
